rclcpp/test/rclcpp/executors/test_static_executor_entities_collector.cpp

```diff
@@ -293,3 +304,319 @@ TEST_F(TestStaticExecutorEntitiesCollector, add_remove_node_with_entities) {
   // Still one for the executor
   EXPECT_EQ(1u, entities_collector_->get_number_of_waitables());
 }
+
+TEST_F(TestStaticExecutorEntitiesCollector, add_callback_group) {
+  auto node = std::make_shared<rclcpp::Node>("node1", "ns");
+  rclcpp::CallbackGroup::SharedPtr cb_group = node->create_callback_group(
```

Comment:
A bit of a nitpick here, but I think using `auto` would let you put these two lines onto one, which would help readability a small bit with all these new tests. Likewise, I would recommend to avoid abbreviations like 'cb_group' if possible (either `group` or `callback_group`). Fix here and below

Reply:
Would that actually, work, though?  If I do that, then the node pointer becomes a temporary, and then wouldn't it be destroyed immediately after this line (since there are no more handles to it)?

Reply:
We discussed this offline.  @brawner was suggesting that I combine this line with the next one by using `auto`.  My personal preference is to not use `auto`, so I think we'll just leave this here for now.  @brawner Let me know if you disagree.

Reply:
Disagreements about the awesomeness and power of auto in a unit test aside, I'm fine leaving these as is.